packages/graphql/lib/src/links/websocket_link/websocket_client.dart

```diff
@@ -95,21 +96,28 @@ class SocketClientConfig {
   ///    return channel;
   /// }
   /// ```
-  ///
-  /// To supply custom headers to an IO client:
-  /// ```dart
-  /// connect: (url, protocols) =>
-  ///   IOWebSocketChannel.connect(url, protocols: protocols, headers: myCustomHeaders)
-  /// ```
-  final WebSocketConnect connect;
-
-  static Future<WebSocketChannel> defaultConnect(
-    Uri uri,
-    Iterable<String>? protocols,
-  ) async {
+  final WebSocketConnect? connectFn;
+
+  /// Custom header to add inside the client
+  final Map<String, dynamic>? headers;
+
+  /// Function to define another connection without call directly
+  /// the connection function
+  FutureOr<WebSocketChannel> connect(
+      {required Uri uri,
+      Iterable<String>? protocols,
+      Map<String, dynamic>? headers,
+      WebSocketConnect? connectFn}) {
+    if (connectFn != null) {
+      return connectFn(uri, protocols);
+    }
+    if (this.connectFn != null) {
+      return this.connectFn!(uri, protocols);
+    }
     return defaultConnectPlatform(
       uri,
       protocols,
+      headers: headers,
```

Comment:
If I'm reading this correctly, the `headers` property is never used. Shouldn't this be something like

```suggestion
      headers: headers ?? this.headers,
```

Reply:
Yes, Good catch thanks